internal/integration/testcase_test.go

```diff
@@ -341,6 +349,57 @@ func (t testcase) AssertCancelled(ctx context.Context, build api.Build) {
 	assert.Equal(t, api.BuildStatesCanceled, t.waitForBuild(ctx, build))
 }
 
+func (t testcase) FirstCommandJobID(build api.Build) string {
+	t.Helper()
+	for _, edge := range build.Jobs.Edges {
+		job, wasJob := edge.Node.(*api.JobJobTypeCommand)
+		if !wasJob {
+			continue
+		}
+		return job.Uuid
+	}
+
+	t.Error("no command job found")
+	return ""
+}
+
+func (t testcase) FailureMessage(build api.Build, jobID string, useStacksAPI bool) string {
+	t.Helper()
+
+	if !useStacksAPI {
+		time.Sleep(10 * time.Second) // Wait for logs to be available
+		return t.FetchLogs(build)
+	}
```

Comment:
Combining with the sleep in FetchLogs, we have 25 seconds wait. Is this necessary?

Reply:
oh good call! i thought i got rid of all the sleeps but i missed the most important one 😅